nodes/arduino_handler.py

```diff
@@ -362,9 +371,12 @@ def process_message(line):
             actuator_state["heater_core_1_1"],
             actuator_state["chiller_compressor_1"]
         ).encode('utf-8')
-        serial_connection.write(message)
-        serial_connection.flush()
-        #rospy.logwarn("arduino_handler serial write {} bytes: >{}<".format(len(message),message.replace('\n','')))
+        try:
+            serial_connection.write(message)
+            serial_connection.flush()
+            trace('arduino_handler serial write %d bytes: >%s<', len(message), message.replace('\n',''))
+        except Exception as e:
+            serial_connection = connect_serial()
 
         # Read from arduino
         buf = serial_connection.readline()
```

Comment:
Also this is from a previous PR but we should check the buffer length and 1st letter inside of process_message since it is already responsible for reading a verifying the message it gets.
I think the second check ("bad bytes") is already done within process message.
The 0 length is also caught by process_message, but I agree we should give it a different error case than the short read.